### PR TITLE
Feature/data structure

### PR DESCRIPTION
**Pull request by emead**
Add data transformation layer to separate REST and DB models.

CPS-247
### Type of change

* [ ] Bug fix
* [ ] New feature
* [ ] Breaking change

### Pull request checklist

* [x] I have added unit tests for new code that I have added
* [ ] I have added/updated functional tests where appropriate
* [x] The code follows our [coding standards](https://github.com/companieshouse/styleguides/blob/master/go.md)

__An exhaustive list of peer review checks can be read [here](https://github.com/golang/go/wiki/CodeReviewComments)__

### CODE REVIEW

File: transformers/payment_test.go [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+				},
+			},
+		}
+		paymentResourceDB := PaymentTransformer{}.TransformToDB(paymentResourceRest)
+		So(paymentResourceDB, ShouldResemble, expectedPaymentResourceDB)
+	})
+}
+
+func TestUnitTransformToRest(t *testing.T) {
+	Convey("Rest converted to DB", t, func() {
```
Comment (macreman) [inline transformers/payment_test.go]: DB converted to Rest ?

File: service/payment.go [hunk 2]
```diff
@@ -111,31 +111,40 @@ func (service *PaymentService) CreatePaymentSession(w http.ResponseWriter, req *
...
 		ID:       req.Header.Get("Eric-Identity"),
 		Email:    email,
 		Forename: forename,
 		Surname:  surname,
 	}
-	paymentResource.Data.Amount = totalAmount
+	paymentResourceRest.Amount = totalAmount
 	// To match the format time is saved to mongo, e.g. "2018-11-22T08:39:16.782Z", truncate the time
-	paymentResource.Data.CreatedAt = time.Now().Truncate(time.Millisecond)
+	paymentResourceRest.Costs = *costs
```
Comment (macreman) [inline service/payment.go]: maybe move this costs assignment down a line after the created at assignment due to the comment above relating to truncating that time

Comment (emead) [reply service/payment.go]: Done

**Comment by macreman** [comment]
looks good 👍

**Pull request merged**